cmd/virt-launcher/virt-launcher.go

```diff
@@ -229,6 +231,7 @@ func main() {
 	readinessFile := flag.String("readiness-file", "/tmp/health", "Pod looks for this file to determine when virt-launcher is initialized")
 	gracePeriodSeconds := flag.Int("grace-period-seconds", 30, "Grace period to observe before sending SIGTERM to vm process")
 	allowEmulation := flag.Bool("allow-emulation", false, "Allow fallback to emulation if /dev/kvm is not present")
+	hookSidecars := flag.Uint("hook-sidecars", 0, "Number of requested hook sidecars, virt-launcher will wait for all of them to become available")
```

Comment:
How are the mapped to the sidecars themselves. 

Say sidecar1 has a bug, and creates `hook1.sock, hook1.2.sock.... hook1.N.sock` consuming all `.sock`ets.

Thinking outloud wouldn't a better alternative would be to have a list of strings of sockets to look for via:

(source: https://stackoverflow.com/questions/28322997/how-to-get-a-list-of-values-into-a-flag-in-golang)

```
package main

import "flag"

type arrayFlags []string

func (i *arrayFlags) String() string {
    return "my string representation"
}

func (i *arrayFlags) Set(value string) error {
    *i = append(*i, value)
    return nil
}

var myFlags arrayFlags

func main() {
    flag.Var(&myFlags, "list1", "Some description for this param.")
    flag.Parse()
}
```

Minor point but would be more reslient, IMO. Though I guess this has drawbacks as well which would be the vm template would have to accept the paths as well..

Just a thought to consider.

Reply:
I was thinking the same, the problem is, that we would need to make template aware of socket names. In the current state, socket name could be arbitrary, we just care about returned values from Info call. I'd rather keep it simple and naive for now.

Reply:
> Say sidecar1 has a bug, and creates hook1.sock, hook1.2.sock.... hook1.N.sock consuming all .sockets.

i'd be okay with just saying that causes undefined behavior for now. To protect users from this, we should at least be explicit in our user documentation that only one socket per a container is allowed though.